articles/kotlin-ranges/index.md

```diff
@@ -0,0 +1,134 @@
+
+In this article, we will take a look at Kotlin ranges.
+
+### What are ranges
+A range is a collection of values determined by a start and stop value. A simple example is, values between 1 and 5 form a range. That is, a range of 1 as the start value and 5 as the end value. Both the start and end values belong to the range as well. In other words, a range is an interval from the start value to end value. We can use functions to create progressions and check for certain values in those ranges. Let's look at how we could apply ranges.  
+
+Before we begin writing code, you can use the following [repl.it](https://repl.it/@Linusmuema/Ranges) to run the functions.
+
+All you need to do is call the specific function in the `main` function. Then run the code and your output will be printed 🤓.
+
+For instance, if we want to run the `printNumRange` function:
+
+```Kotlin
+  fun main(args: Array<String>) {
+      // call the function here
+      printNumRange()
+  }
+```
+
+### 1. To Create a Progression
+We can use the range operator to create an iterable range. A loop is created that goes through each item from the start value to the end value. This helps us to be sure of the number of iterations in the loop. It gives the developer more control over the loop. In the repl.it, call the `printNumRange` function and run it. The output should be:
+
+```bash
+1,2,3,4,5,
+```
+
+This loop goes from 1 to 5. That's not the only trick ranges can do. We can also create ranges using characters. We can create a range that starts from a specific letter in the alphabet to another. This can be from `a` to `e`. You can go ahead and call the `printCharRange` function.
+
+The expected output is:
+
+```bash
+a,b,c,d,e,
+```
+It was not easy to do this in java. When implemented it led to lots of boilerplate and dirty code.
+
+```Java
+  String letters = "abcdefghijklmopqrstuvwxyz'";
+  for (char c : letters.toCharArray()) {
+      System.out.printf(c);
+  }
+```
+
+First we create a string of the characters. Then we convert it to a CharArray where using `forEach`, we get each of the characters. With Kotlin ranges, this is reduced to only a few lines with fewer requirements.
+
+### 2. To Create Reverse Loops
+It has always been a challenge to create a reverse loop, as a developer would have interchange values the old way. This could lead to confusion within the code and it would take time to debug.
+
+An example of a reverse loop in Java would be:
+
+```Java
+for (int i = 5; i >= 0; i--) {}
+```
+
+Ranges have a function called `downTo` that can be used to create a reverse progression. We can use it to loop from the end value backwards to the start point.
+
+Once you run the `reverseNumRange`, the output should be:
+
+```bash
+5,4,3,2,1,
+```
+
+The same applies to the character range. It loops downwards to the start character. The function `reverseCharRange` does this for us.
+
```

Comment:
Please include the definition of `reverseCharRange` function here.

Reply:
added the function definition